torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -339,7 +412,14 @@ void runTestKernel(
       TORCH_INTERNAL_ASSERT(
           input.toTensor().device().index() == entry->device_,
           "input to kernel on device that is not compiled for");
-      kernel_args.push(input.toTensor(), outputs[0].sizes());
+      TORCH_INTERNAL_ASSERT(
+          !entry->outputs.empty(),
+          "No output found for this kernel, aborting.");
+      if (has_reduction) {
+        kernel_args.push(input.toTensor());
```

Comment:
So reduction & limited broadcast are exclusive to each other at the moment?

Reply:
@kevinstephano likely you will want to do the same thing for `runTestKernel`.
Maybe we should merge code given that the test API is not going away for a while.

Reply:
I am pretty sure that runTestKernel and runKernel are pretty close to mergeable as I started to use IValues in runTestKernel.

Reply:
I asked  @naoyam to look at better error handling for runTestKernel: https://github.com/csarofeen/pytorch/pull/30 still some work to be done there.